GTG/backends/backend_caldav.py

```diff
@@ -0,0 +1,831 @@
+                    self.sync_engine_url.record_relationship(
+                        local_id=tid, remote_id=str(todo.url),
+                        meme=meme)
+                    self._add_to_cache(todo)
+
+            elif action == SyncEngine.UPDATE:
+                with self.datastore.get_backend_mutex():
+                    todo = self._get_todo(url) # network lookup
+                    meme = self.sync_engine_url.get_meme_from_local_id(task.get_id())
+                    newest = meme.which_is_newest(task.get_modified(),
+                                                  todo.get_modified() or Date.no_date())
+                    if newest == "local":
+                        self._populate_todo(task, todo, save_new=False)
+                        meme.set_remote_last_modified(todo.get_modified())
+                        meme.set_local_last_modified(task.get_modified())
+                    else:
+                        # we skip saving the state
+                        return
+
+            elif action == SyncEngine.REMOVE:
+                self.datastore.request_task_deletion(tid)
+                try:
+                    self.sync_engine_url.break_relationship(local_id=tid)
+                except KeyError:
+                    pass
+
+            elif action == SyncEngine.LOST_SYNCABILITY:
+                todo = self._get_todo(url) # network lookup TODO: try to avoid it
+                self._exec_lost_syncability(tid, todo)
+
+            self.save_state()
+
+    def _populate_todo(self, task, todo, save_new=None):
+        # Title
+        todo.set_title(task.get_title())
+
+        # Text
+        text = task.get_excerpt(strip_tags=True, strip_subtasks=True)
+        if todo.get_text() != text:
+            todo.set_text(text)
+
+        # Parents
+        parents = task.get_parents()
+        todo.set_parents(parents, to_id=lambda tid: self._tid_to_id(tid))
+
+        # dates
+        todo.set_modified(task.get_modified())
+        todo.set_due_date(task.get_due_date())
+        todo.set_added_date(task.get_added_date())
+        todo.set_closed_date(task.get_closed_date())
+        todo.set_start_date(task.get_start_date())
+        todo.set_modified(task.get_modified())
+
+        # Status
+        status = task.get_status()
+        if _CALDAV_TO_GTG_STATUS[todo.get_vstatus()] != status:
+            todo.set_status(status)
+
+        # Tags
+        cal_name = todo.parent_name
+        gtg_tags = set([self._tag_from_gtg(tag.get_name()) for tag in task.get_tags()])
+        todo_tags = set(todo.get_tags())
+        # tags to add
+        for tag in gtg_tags.difference(todo_tags):
+            if tag != cal_name:
+                todo.add_tag(tag)
+        # tags to remove
+        for tag in todo_tags.difference(gtg_tags):
+            todo.remove_tag(tag)
+
+
+        try:
+            todo.save(save_new=save_new)
+        except:
+            log.debug(f'_populate_todo todo.instance={todo.instance}')
```

Comment:
Are there specific errors that can be caught rather than a bare except? Also the debug could be a bit more informative ("Todo saving failed for instance x" or something like that)